ci-operator/step-registry/cucushift/installer/check/ibmcloud/custom-endpoints/cucushift-installer-check-ibmcloud-custom-endpoints-commands.sh

```diff
@@ -28,11 +15,100 @@ function get_service_endpoint() {
     echo "${service_endpoint}"
 }
 
+function exists_in()
+{
+	local val=$1
+	shift
+	local val_list=("$@")
+	local match=1
+	for v in "${val_list[@]}";
+	do
+	    if [[ "$val" = "$v" ]]; then
+	        match=0
+	        break
+	    fi
+	done
+	return $match
+}
+
+function isPreVersion() {
+  local required_ocp_version="$1"
+  local isPre
+  version=$(oc version -o json | jq -r '.openshiftVersion' | cut -d '.' -f1,2)
+  echo "get ocp version: ${version}"
+
+  isPre=0
+  if [ -n "${version}" ] && [ "$(printf '%s\n' "${required_ocp_version}" "${version}" | sort --version-sort | head -n1)" = "${required_ocp_version}" ]; then
+    isPre=1
+  fi
+  return $isPre
+}
+
+function check_ep_names() {
+    local isPreVersion="$1"
+    local output expString
+    output=$(openshift-install explain installconfig.platform.ibmcloud.serviceEndpoints)
+    if [[ "${isPreVersion}" == "True" ]]; then
+        expString='Valid Values: "COS","DNSServices","GlobalCatalog","GlobalSearch","GlobalTagging","IAM","KeyProtect","ResourceController","ResourceManager","VPC"'
+    else
+        expString='Valid Values: "CIS","COS","COSConfig","DNSServices","GlobalCatalog","GlobalSearch","GlobalTagging","HyperProtect","IAM","KeyProtect","ResourceController","ResourceManager","VPC"'
+    fi
+    if ! [[ ${output} == *"${expString}"* ]]; then
+        echo "ERROR: Unexpected explain installconfig.platform.ibmcloud.serviceEndpoints: - ${output} "
+        return 1
+    else
+        echo "Check installconfig.platform.ibmcloud.serviceEndpoints passed."
+        return 0
+    fi
+}
+
+function check_eps() {
+    local isPre418="$1"
+    local epFile url ret srv_418 src_name service_endpoint url
+    ret=0
+    epFile="${ARTIFACT_DIR}/eps.json"
+    oc get infrastructure -o json | jq -r '.items[0].status.platformStatus.ibmcloud.serviceEndpoints' > ${epFile}
+    declare -a srv_array
+    readarray -t srv_array < <(jq -r 'keys[]' ${DEFAULT_PRIVATE_ENDPOINTS})
+    srv_418=("COSConfig" "GlobalCatalog")
```

Comment:
https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/58984/rehearse-58984-periodic-ci-openshift-verification-tests-master-installer-rehearse-4.17-installer-rehearse-ibmcloud-dis/1862360731010207744 
How about keep it to avoid the 

> WARN:  No custom endpoint is defined for GlobalCatalog
> WARN:  No custom endpoint is defined for COSConfig